### ISSUE TITLE
Forbid to use methods without arguments

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1083]
# Rule request

## Thesis

You can write this code:

```python
class Test(object):
     def some(self):
           return 42
```

This indicates a design problem:
- It can be a `@property`
- It can be a class-level variable / constant
- It can be an instance level variable

So, the rule is: when we see a method without `@property` decorator with zero argument (we do not count `self`, `cls`, and `mcs` here) - we raise a violation.

It is related to `oop` group.

**Comment by sobolevn** [comment]
Related: https://docs.quantifiedcode.com/python-anti-patterns/correctness/method_has_no_argument.html

**Comment by AlwxSin** [comment]
Disagree. There is a lot of examples how to use methods without arguments.
```python
class AsyncDbConnector(object):
    
    def __init__(self, connection_string):
        self.connection_string = connection_string
        
    async def connect(self):
        """It can be in __aenter__ method, but for example"""
        await self.on_startup()
        
    async def on_startup(self):
        """Do some stuff before connect."""
        
    async def on_exit(self):
        """"""
   async def ping(self):
        """"""
```

**Comment by sobolevn** [comment]
Yeap, I agree with you. There are multiple valid cases when methods without arguments are a good thing: life-cylce hooks, fabrics, iterators, convertors, etc.
